SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialInterestConstraints.h

```diff
@@ -0,0 +1,270 @@
+
+	virtual void CreateConstraint(const USpatialClassInfoManager& ClassInfoManager, SpatialGDK::QueryConstraint& OutConstraint) const override;
+
+	/** The size of the cylinder represented by this constraint in centimeters. */
+	UPROPERTY(BlueprintReadOnly, EditDefaultsOnly, Meta = (ClampMin = 0.0))
+	float Radius = 0.0f;
+};
+
+/**
+ * Creates a constraint that includes all entities within a bounding box centered on the actor.
+ */
+UCLASS(BlueprintType, EditInlineNew, DefaultToInstanced)
+class SPATIALGDK_API URelativeBoxConstraint final : public UAbstractQueryConstraint
+{
+	GENERATED_BODY()
+public:
+	URelativeBoxConstraint() = default;
+	~URelativeBoxConstraint() = default;
+
+	virtual void CreateConstraint(const USpatialClassInfoManager& ClassInfoManager, SpatialGDK::QueryConstraint& OutConstraint) const override;
+
+	/** The size of the box represented by this constraint. */
+	UPROPERTY(BlueprintReadOnly, EditDefaultsOnly, Meta = (ClampMin = 0.0))
+	FVector EdgeLengths = FVector::ZeroVector;
+};
+
+/**
+ * Creates a constraint that includes an actor type (including subtypes) and a cylindrical range around the actor with the interest.
+ */
+UCLASS(BlueprintType, EditInlineNew, DefaultToInstanced)
+class SPATIALGDK_API UCheckoutRadiusConstraint final : public UAbstractQueryConstraint
+{
+	GENERATED_BODY()
+public:
+	UCheckoutRadiusConstraint() = default;
+	~UCheckoutRadiusConstraint() = default;
+
+	virtual void CreateConstraint(const USpatialClassInfoManager& ClassInfoManager, SpatialGDK::QueryConstraint& OutConstraint) const override;
+
+	/** The base type of actor that this constraint will capture. */
+	UPROPERTY(BlueprintReadOnly, EditDefaultsOnly)
+	TSubclassOf<AActor> ActorClass;
+
+	/** The size of the cylinder represented by this constraint in centimeters. */
+	UPROPERTY(BlueprintReadOnly, EditDefaultsOnly, Meta = (ClampMin = 0.0))
+	float Radius = 0.0f;
+};
+
+/**
+ * Creates a constraint that includes all actors of a type (optionally including subtypes).
+ */
+UCLASS(BlueprintType, EditInlineNew, DefaultToInstanced)
+class SPATIALGDK_API UActorClassConstraint final : public UAbstractQueryConstraint
+{
+	GENERATED_BODY()
+public:
+	UActorClassConstraint() = default;
+	~UActorClassConstraint() = default;
+
+	virtual void CreateConstraint(const USpatialClassInfoManager& ClassInfoManager, SpatialGDK::QueryConstraint& OutConstraint) const override;
+
+	/** The base type of actor that this constraint will capture. */
+	UPROPERTY(BlueprintReadOnly, EditDefaultsOnly)
+	TSubclassOf<AActor> ActorClass;
+
+	/** Whether this constraint should capture derived types. */
+	UPROPERTY(BlueprintReadOnly, EditDefaultsOnly)
+	bool bIncludeDerivedClasses = true;
+};
+
+/**
+ * Creates a constraint that includes all components of a type (optionally including subtypes).
+ */
+UCLASS(BlueprintType, EditInlineNew, DefaultToInstanced)
+class SPATIALGDK_API UComponentClassConstraint final : public UAbstractQueryConstraint
```

Comment:
What's the reason for separating these Actor and Component? I think combining them with a UClass is fine.

Reply:
Even though the back-end constraints are basically the same, I think the unreal interface is better if they're split - the dropdown is populated with Actor classes for the Actor constraint, and component classes for the component constraint.

Reply:
I don't even think we want a generic UClass version, at least not unless we have a concrete use case to justify that.